File: src/main/java/dev/abarmin/aml/dashboard/model/BlockTypeModel.java
package dev.abarmin.aml.dashboard.model;

import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class BlockTypeModel {
  private String name;
  private String icon;
  private String type;
  private String previewComponent;
  private String configComponent;
}

File: aml-frontend-private/src/model/BlockTypeModel.ts
export type BlockType = "AVATAR_BLOCK" | "BUTTON_BLOCK" | "HEADER_BLOCK" | "SOCIAL_NETWORKS_BLOCK";

export class BlockTypeModel {
    name: string;
    icon: string;
    type: BlockType;
    previewComponent: string;
    configComponent: string;

    constructor(
        name: string,
        icon: string,
        type: BlockType,
        previewComponent: string,
        configComponent: string
    ) {
        this.name = name;
        this.icon = icon;
        this.type = type;
        this.previewComponent = previewComponent;
        this.configComponent = configComponent;
    }
}
